apps/spark-3.5/src/test/java/com/linkedin/openhouse/catalog/e2e/SparkMoRFunctionalTest.java

```diff
@@ -0,0 +1,265 @@
+        tableName);
+  }
+
+  protected void createAndInitTable(String schema) {
+    sql("CREATE TABLE openhouse.%s (%s) USING iceberg %s", tableName, schema, "");
+    initTable();
+  }
+
+  @Test
+  public void testDeleteFilesCanBeCreated() throws NoSuchTableException {
+    createAndInitTable("id int, data string");
+
+    List<SimpleRecord> records =
+        Arrays.asList(
+            new SimpleRecord(1, "a"),
+            new SimpleRecord(1, "b"),
+            new SimpleRecord(1, "c"),
+            new SimpleRecord(2, "d"),
+            new SimpleRecord(2, "e"));
+    ops.spark()
+        .createDataset(records, Encoders.bean(SimpleRecord.class))
+        .coalesce(1)
+        .writeTo(tableName)
+        .append();
+
+    sql("DELETE FROM %s WHERE id = 1 and data='a'", tableName);
+    sql("DELETE FROM %s WHERE id = 2 and data='d'", tableName);
+    sql("DELETE FROM %s WHERE id = 1 and data='c'", tableName);
+
+    IcebergTableStats stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles() + stats.getNumEqualityDeleteFiles()).isEqualTo(3L);
+
+    List<Object[]> expected = Arrays.asList(row(1, "b"), row(2, "e"));
+    List<Object[]> actual = sql("SELECT * FROM %s ORDER BY id ASC", tableName);
+    assertThat(actual).containsExactlyElementsOf(expected);
+  }
+
+  @Test
+  public void testCompactionCanRemovePositionDeleteFiles() throws NoSuchTableException {
+    createAndInitTable("id int, data string");
+
+    List<SimpleRecord> records =
+        Arrays.asList(
+            new SimpleRecord(1, "a"),
+            new SimpleRecord(1, "b"),
+            new SimpleRecord(1, "c"),
+            new SimpleRecord(2, "d"),
+            new SimpleRecord(2, "e"));
+    ops.spark()
+        .createDataset(records, Encoders.bean(SimpleRecord.class))
+        .coalesce(1)
+        .writeTo(tableName)
+        .append();
+
+    sql("DELETE FROM %s WHERE id = 1 and data='a'", tableName);
+    sql("DELETE FROM %s WHERE id = 2 and data='d'", tableName);
+    sql("DELETE FROM %s WHERE id = 1 and data='c'", tableName);
+
+    Table table = ops.getTable(tableName);
+
+    // first run of compaction
+    RewriteDataFiles.Result result = rewriteFunc.apply(ops, table);
+    Assertions.assertEquals(1, result.addedDataFilesCount());
+    Assertions.assertEquals(1, result.rewrittenDataFilesCount());
+
+    // this asserts the dangling delete problem, since compaction did NOT remove the final delete
+    IcebergTableStats stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles()).isEqualTo(3L);
+    assertThat(stats.getNumCurrentSnapshotPositionDeleteFiles()).isEqualTo(1L);
+
+    // notice that running compaction again with no new commits cleans up the dangling delete
+    rewriteFunc.apply(ops, table);
+    stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles()).isEqualTo(3L);
```

Comment:
this represents the number of pos-deletes across the whole tables' history?

Reply:
yes. as opposed to the current snapshot metrics will only have latest table history